### DISCUSSION TITLE
Persistence to a file (xml or json)

### DISCUSSION THREAD

**Discussion by ggo98** [quartznet/quartznet#1817] [Q&A] [answered]
Hello,

I would like to save my jobs and triggers to an XML file.
I've been unable to find the solution in one day.

Maybe someone knows how to do that...

Thank you,
best regards,
Olivier gg.

**Comment by lahma** [comment, marked as answer]
Here's the sample for XML configuration: https://github.com/quartznet/quartznet/blob/main/src/Quartz.Examples/10_RunningJobsByPlugInXmlConfiguration/RunningJobsByPlugInXmlConfigurationExample.cs

And then the actual XML file: https://github.com/quartznet/quartznet/blob/main/src/Quartz.Examples/quartz_jobs.xml

**Reply by ggo98** [reply]
Thank you so much :)
Shame on me... I had not seen this example...
Best regards,
Olivier.

**Comment by ggo98** [comment]
Hello,

actually this example doesn't demonstrate exactly what I would like to do...
Is it possible to create a job programmatically (with 
IJobDetail job = JobBuilder.Create<HelloJob>()....Build(),
ITrigger trigger = TriggerBuilder.Create().....Build(),
and await sched.ScheduleJob(job, trigger);)
, and save the "HelloJob" definition in the quartz_jobs.xml file please ?
Is it possible ?

Thank you,
Olivier.

**Reply by lahma** [reply]
You can generate the XML file by yourself using .NET XML APIs I believe. You should be able to instantiate `QuartzXmlConfiguration20`, fill it with objects you need and then serialize it to file.

**Reply by ggo98** [reply]
Ok, I'll try this. Thanks a lot.
Best regards,
Olivier.
